src/averageAges.js

```diff
@@ -15,11 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeople = people.filter(person => {
+    return century
+      ? Math.ceil(person.died / 100) === century
+      && person.sex === 'm'
+      : person.sex === 'm';
+  });
+
+  const peopleAges = filteredPeople.map((person) => {
+    return person.died - person.born;
+  });
+
+  const averageAge = peopleAges.reduce((prevPerson, nextPerson) => {
+    return prevPerson + nextPerson;
+  }, 0) / peopleAges.length;
```

Comment:
You are using the similar algorithm for men and women. So you can make one function for both options.

Reply:
Good idea, thanks!